src/foundry/client/pixi/core/containers/quadtree.d.ts

```diff
@@ -1,166 +1,198 @@
+    /**
+     * Get all the objects which could collide with the provided rectangle
+     * @param rect - The normalized target rectangle
+     * @param options - Options affecting the collision test.
+     * @returns The objects in the Quadtree which represent potential collisions
+     */
+    getObjects(
+      rect: Rectangle,
+      options?: {
+        /* Function to further refine objects to return
+         * after a potential collision is found. Parameters are the object and rect, and the
+         * function should return true if the object should be added to the result set.
+         */
+        collisionTest: (o: QuadtreeObject<T>, rect: Rectangle) => boolean;
+
+        /** The existing result set, for internal use.
+         *  (default: `new Set<T>()`)
+         */
+        _s?: Set<T>;
+      },
+    ): Set<T>;
 
-  /**
-   * Visualize the nodes and objects in the quadtree
-   * @param objects - Visualize the rectangular bounds of objects in the Quadtree. Default is false.
-   *                  (default: `false`)
-   */
-  visualize({ objects }?: { objects?: boolean }): void;
-}
+    /**
+     * Obtain the leaf nodes to which a target rectangle belongs.
+     * This traverses the quadtree recursively obtaining the final nodes which have no children.
+     * @param rect - The target rectangle.x
+     * @returns The Quadtree nodes to which the target rectangle belongs
+     */
+    getLeafNodes(rect: Rectangle): Quadtree<T>[];
 
-declare namespace Quadtree {
-  /**
-   * Additional options which configure the Quadtree
-   */
-  interface Options<T> {
     /**
-     * The maximum number of objects per node
-     * @defaultValue `20`
+     * Obtain the child nodes within the current node which a rectangle belongs to.
+     * Note that this function is not recursive, it only returns nodes at the current or child level.
+     * @param rect - The target rectangle.
+     * @returns The Quadtree nodes to which the target rectangle belongs
      */
-    maxObjects?: number;
+    getChildNodes(rect: Rectangle): Quadtree<T>[];
 
     /**
-     * The maximum number of levels within the root Quadtree
-     * @defaultValue `4`
+     * Identify all nodes which are adjacent to this one within the parent Quadtree.
      */
-    maxDepth?: number;
+    getAdjacentNodes(): Quadtree<T>[];
 
     /**
-     * The depth level of the sub-tree. For internal use
-     * @defaultValue `0`
-     * @internal
+     * Visualize the nodes and objects in the quadtree
+     * @param objects - Visualize the rectangular bounds of objects in the Quadtree. Default is false.
+     *                  (default: `false`)
      */
-    _depth?: number;
+    visualize({ objects }?: { objects?: boolean }): void;
+  }
+
+  /**
+   * A subclass of Quadtree specifically intended for classifying the location of objects on the game canvas.
+   */
+  class CanvasQuadtree extends Quadtree<object> {
+    // @ts-expect-error Override monkeypatch executed in class constructor
```

Comment:
What exactly does this mean? Can you show the code you're talking about in the constructor? And the error without `@ts-expect-error`?

Reply:
`'bounds' is defined as a property in class 'Quadtree<object>', but is overridden here in 'CanvasQuadtree' as an accessor.`

I'm now realizing that it might be simpler to do `readonly bounds: SceneDimensions["rect"];`

Reply:
The code is `Object.defineProperty(this, "bounds", {get: () => canvas.dimensions.rect});`